app/src/main/java/com/github/se/cyrcle/ui/review/ReviewScreen.kt

```diff
@@ -0,0 +1,125 @@
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center) {
+              // Stars reflecting the slider value (supports half stars)
+              Row(
+                  modifier = Modifier.padding(8.dp).testTag("StarsRow"),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    for (i in 1..5) {
+                      val starIcon =
+                          when {
+                            i <= sliderValue -> Icons.Filled.Star // Full star
+                            i - 0.5f == sliderValue -> Icons.Filled.StarHalf // Half star
+                            else -> Icons.Filled.StarBorder // Empty star
+                          }
+                      Icon(
+                          imageVector = starIcon,
+                          contentDescription = null,
+                          tint = MaterialTheme.colorScheme.primary,
+                          modifier = Modifier.size(40.dp).testTag("Star$i"))
+                    }
+                  }
+
+              // Display the experience text based on the slider value
+              Text(
+                  text =
+                      when (sliderValue) {
+                        in 0f..1f -> "Terrible experience!"
+                        in 1f..2f -> "Bad experience :("
+                        in 2f..3f -> "Poor experience :/"
+                        3f -> "Average experience..."
+                        in 3.5f..4f -> "Good experience!"
+                        in 4.5f..5f -> "Great experience!"
+                        else -> ""
+                      },
+                  style = MaterialTheme.typography.bodyLarge,
+                  modifier = Modifier.padding(top = 8.dp).testTag("ExperienceText"))
+
+              // Slider with step granularity of 0.5
+              Text(text = "Rating: $sliderValue", style = MaterialTheme.typography.bodyLarge)
+              Slider(
+                  value = sliderValue,
+                  onValueChange = { newValue -> sliderValue = newValue },
+                  valueRange = 0f..5f,
+                  steps = 9, // Steps for granularity of 0.5 between 0 and 5
+                  modifier = Modifier.padding(16.dp).testTag("Slider"))
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              // OutlinedTextField for text input
+              OutlinedTextField(
+                  value = textValue,
+                  onValueChange = { newText -> textValue = newText },
+                  label = { Text("Write your review") },
+                  modifier = Modifier.fillMaxWidth().testTag("ReviewInput"))
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              // Add Review Button
+              Button(
+                  onClick = {
+                    Toast.makeText(context, "Review Added!", Toast.LENGTH_SHORT).show()
+                    navigationActions.navigateTo(Screen.CARD)
```

Comment:
This should be a `navigationActions.goBack()` otherwise the go back of the CardScreen leads back to the ReviewScreen (the navigation stack is not poped)

Reply:
Fixed in commit!